Core/NoOpRunner.Core/Shapes/EntityShape.cs

```diff
@@ -0,0 +1,23 @@
+﻿using NoOpRunner.Core.Entities;
+using System.Diagnostics;
+
+namespace NoOpRunner.Core.Shapes
+{
+    public abstract class EntityShape : BaseShape
+    {
+        public EntityShape(int centerPosX, int centerPosY) : base(centerPosX, centerPosY)
+        {
+
+        }
+
+        public override bool CanOverlap(BaseShape other)
```

Comment:
Because the child classes of `EntityShape` do not really have a common implementation for either `CanOverlap` or `OnCollision`, do not implement the methods here (`EntityShape` is abstract anyways, you don't need to). The justification for not inheriting directly from BaseShape can be something like different generation rules in the future for Entities

Reply:
Fixed